### ISSUE TITLE
Error: no SSL certificate specified for bind '0.0.0.0:443'

### ISSUE THREAD

**Issue by ohardy** [haproxytech/kubernetes-ingress#166]
Hi

Step to reproduce the bug:
- Start the ingress controller (without TLS configuration) with no ingresses in the cluster or with ingresses without any TLS.
- Add an ingress with TLS secret
- The bug appear, certificates will never be loaded in the ingress controller pod (Haproxy configurations appear in `/tmp/haproxy/failed` folder).
- Delete the pod, when a new ingress controller start, everything is ok.

Log:
```
2020/03/06 12:13:51 HAProxy Ingress Controller v1.3.3 579bc27

2020/03/06 12:13:51 Build from: git@github.com:haproxytech/kubernetes-ingress.git
2020/03/06 12:13:51 Build date: 2020-03-04T08:11:52

2020/03/06 12:13:51 ConfigMap: public-haproxy-ingress/haproxy-def
2020/03/06 12:13:51 Ingress class: haproxy
2020/03/06 12:13:51 Publish service: 
2020/03/06 12:13:51 Default backend service: public-haproxy-ingress/ingress-default-backend
2020/03/06 12:13:51 Default ssl certificate: public-haproxy-ingress/
2020/03/06 12:13:51 Running with  HA-Proxy version 2.0.13 2020/02/13 - https://haproxy.org/
2020/03/06 12:13:51 Starting HAProxy with /etc/haproxy/haproxy.cfg
2020/03/06 12:13:51 Running on haproxy-ingress-557fb6568b-qg7dg
[WARNING] 065/121351 (20) : Can't open server state file '/var/state/haproxy/global': No such file or directory
2020/03/06 12:13:51 Running on Kubernetes version: v1.16.6-gke.4 linux/amd64
[NOTICE] 065/121351 (22) : New worker #1 (23) forked
2020/03/06 12:13:57 Confiugring default_backend ingress-default-backend from ingress DefaultService
2020/03/06 12:13:57 HAProxy reloaded
[NOTICE] 065/121357 (30) : New worker #1 (31) forked
2020/03/06 13:26:58 HAProxy reloaded
[NOTICE] 065/132658 (39) : New worker #1 (40) forked
2020/03/06 13:29:52 controller/haproxy.go:149 14: ERR transactionId=3975052f-68be-4849-8b4a-c89e3c589b83 
msg="Proxy 'https': no SSL certificate specified for bind '0.0.0.0:443' at [/tmp/haproxy/haproxy.cfg.3975052f-68be-4849-8b4a-c89e3c589b83:51] (use 'crt')."
msg="Proxy 'https': no SSL certificate specified for bind ':::443' at [/tmp/haproxy/haproxy.cfg.3975052f-68be-4849-8b4a-c89e3c589b83:52] (use 'crt')."
2020/03/06 13:29:52 monitor.go:120: 14: ERR transactionId=3975052f-68be-4849-8b4a-c89e3c589b83 
msg="Proxy 'https': no SSL certificate specified for bind '0.0.0.0:443' at [/tmp/haproxy/haproxy.cfg.3975052f-68be-4849-8b4a-c89e3c589b83:51] (use 'crt')."
msg="Proxy 'https': no SSL certificate specified for bind ':::443' at [/tmp/haproxy/haproxy.cfg.3975052f-68be-4849-8b4a-c89e3c589b83:52] (use 'crt')."
2020/03/06 13:29:57 haproxy.go:157: HAProxy reloaded
[NOTICE] 065/132957 (48) : New worker #1 (49) forked
2020/03/06 13:31:14 haproxy.go:157: HAProxy reloaded
[NOTICE] 065/133114 (56) : New worker #1 (57) forked
2020/03/06 13:41:50 controller/haproxy.go:149 14: ERR transactionId=0ca81b35-3e64-4f2d-89ee-6e327d1330bc 
msg="Proxy 'https': no SSL certificate specified for bind '0.0.0.0:443' at [/tmp/haproxy/haproxy.cfg.0ca81b35-3e64-4f2d-89ee-6e327d1330bc:51] (use 'crt')."
msg="Proxy 'https': no SSL certificate specified for bind ':::443' at [/tmp/haproxy/haproxy.cfg.0ca81b35-3e64-4f2d-89ee-6e327d1330bc:52] (use 'crt')."
2020/03/06 13:41:50 monitor.go:120: 14: ERR transactionId=0ca81b35-3e64-4f2d-89ee-6e327d1330bc 
msg="Proxy 'https': no SSL certificate specified for bind '0.0.0.0:443' at [/tmp/haproxy/haproxy.cfg.0ca81b35-3e64-4f2d-89ee-6e327d1330bc:51] (use 'crt')."
msg="Proxy 'https': no SSL certificate specified for bind ':::443' at [/tmp/haproxy/haproxy.cfg.0ca81b35-3e64-4f2d-89ee-6e327d1330bc:52] (use 'crt')."
2020/03/06 13:41:55 haproxy.go:157: HAProxy reloaded
[NOTICE] 065/134155 (65) : New worker #1 (66) forked
```
Thanks

**Comment by Mo3m3n** [comment]
Thanks @ohardy for reporting this.
I have pushed a fix to a similar issue if not the same: https://github.com/haproxytech/kubernetes-ingress/pull/167
You can give the dev image a try `haproxytech/kubernetes-ingress:dev` until the next release is available later this week.

**Comment by ohardy** [comment]
It's fixed in the dev image, thanks !

**Comment by Mo3m3n** [comment]
Released in `haproxytech/kubernetes-ingress:1.4.0`
